src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1075,11 +1075,10 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                 methods = constructedMethods.ToImmutableAndFree();
             }
 
-            if (methods.Length == 1 && !extensionMethodsOfSameViabilityAreAvailable)
+            if (methods.Length == 1 && methods[0].Arity == 0 && !extensionMethodsOfSameViabilityAreAvailable)
```

Comment:
I will add an explanatory comment, but I don't think it will be valuable to add comments that explain why the previous code was wrong.

---
In reply to: [67029427](https://github.com/dotnet/roslyn/pull/11981#discussion_r67029427) [](ancestors = 67029427,67028724,67027973,67027377,67019197)